bin/mag.js

```diff
@@ -0,0 +1,88 @@
+
+db.init(function(err) {
+    if(err) throw err;
+    rcon = redis.createClient(config.redis.port, config.redis.server);
+    rcon.on('error', err=>{throw err});
+    rcon.on('ready', ()=>{
+        logger.info("connected to redis");
+        setInterval(health_check, 1000*60*2); //start checking health 
+        setTimeout(health_check, 1000*10); //run shortly after start
+        run();
+    });
+});
+
+var report = {
+    status: "ok",
+    maxage: 1000*60*30,
+    project_counts: 0,
+}
+
+function health_check() {
+    report.date = new Date();
+    rcon.set("health.warehouse.projectinfo."+process.env.HOSTNAME+"-"+process.pid, JSON.stringify(report));
+}
+
+function run() {
+	db.Projects.find({
+        removed: false,
+    })
+    //.populate('app project')
+    .exec((err, projects)=>{
+		if(err) throw err;
+        report.project_counts = projects.length;
+        async.eachSeries(projects, handleProject, err=>{
+            if(err) logger.error(err);
+            console.log("done going through all projects sleeping.....");
+            setTimeout(run, 1000*3600*3);
+        });
+	});
+}
+
+
+function generateQuery(str){
+    let fil = str.toLowerCase()
+    .split(' ').
+    filter((word) => !stopwords.includes(word)).
+    map((word) => `W='${word}'`).join();
+    return "OR("+fil+")"
+}
+
+
+
+function handleProject(app, cb) {
+    logger.debug("....................... %s %s", app.name, app._id.toString());
+
+    str = ""+app.name+" "+app.desc
+
+    query = generateQuery(str)
```

Comment:
You can/should move generateQuery to be part of the updateProjectMag() and hosted in common.